.github/workflows/create-release-pr.yml

```diff
@@ -0,0 +1,214 @@
+name: Create Release PR
+
+on:
+  workflow_dispatch:
+    inputs:
+      release_version:
+        description: 'The release_version used for the release branch name, e.g. release/vx.x.x'
+        default: 'vx.x.x'
+        required: true
+        type: string
+      pre_release_version:
+        description: "Pre-Release version, e.g. 'beta.1', will be added behind the release_version as the tag."
+        required: false
+        type: string
+
+env:
+  RELEASE_VERSION: ${{ inputs.release_version }}
+  PRE_RELEASE_VERSION: ${{ inputs.pre_release_version }}
+  RELEASE_BRANCH: release/${{ inputs.release_version }}
+
+jobs:
+  create-release-pr:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Set Release Version and Branch to Check Out
+        id: set-release
+        run: |
+          if [[ $RELEASE_VERSION =~ ^v[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
+            if [[ $PRE_RELEASE_VERSION =~ ^[-a-z.0-9]+$ ]]; then
```

Comment:
Is this supposed to have the `-` before the a?

Reply:
I guess that just allows dashes?

Reply:
@onetechnical Yeah it's to allow hyphens since go sdk uses hyphens for the pre-release version. So the version would be x.x.x-beta.1 The dash has to be in the front or back of the regex

Reply:
Hmm... although technically I don't expect the pre_release_version to have the hyphen. I will join the `${RELEASE_VERSION}-${PRE_RELEASE_VERSION}`

Reply:
Maybe I should remove allowing a hyphen.